sdk/core/azure-core/azure/core/tracing/_inference_api_injector.py

```diff
@@ -0,0 +1,370 @@
+        else:
+            response: dict[str, Any] = {
+                "finish_reason": str(choice.finish_reason),
+                "index": choice.index,
+            }
+            attributes={
+                "get_ai.system": "openai",
+            }
+        if choice.message.tool_calls:
+            response["message"]["tool_calls"] = [tool.as_dict() for tool in choice.message.tool_calls]
+        span.span_instance.add_event(name="gen_ai.choice", attributes=attributes)
+
+
+def _add_response_chat_attributes(span: AbstractSpan, result: _models.ChatCompletions | _models.StreamingChatCompletionsUpdate) -> None:
+    _set_attributes(
+        span,
+        ("gen_ai.response.id", result.id),
+        ("gen_ai.response.model", result.model),
+        ("gen_ai.response.finish_reason", str(result.choices[-1].finish_reason)),
+        ("gen_ai.usage.completion_tokens", result.usage.completion_tokens if hasattr(result, "usage") and result.usage else None),
+        ("gen_ai.usage.prompt_tokens", result.usage.prompt_tokens if hasattr(result, "usage") and result.usage else None),
+    )
+
+
+def _add_request_span_attributes(span: AbstractSpan, span_name: str, kwargs: Any) -> None:
+    global _trace_inference_content
+    if span_name.startswith("ChatCompletionsClient.complete"):
+        _add_request_chat_attributes(span, **kwargs)
+        if _trace_inference_content:
+            _add_request_chat_message_event(span, **kwargs)
+    # TODO add more models here
+
+
+def _add_response_span_attributes(span: AbstractSpan, result: object) -> None:
+    if isinstance(result, _models.ChatCompletions):
+        _add_response_chat_attributes(span, result)
+        _add_response_chat_message_event(span, result)
+    # TODO add more models here
+
+
+def _accumulate_response(item, accumulate: dict[str, Any]) -> None:
+    if item.finish_reason:
+        accumulate["finish_reason"] = item.finish_reason
+    if item.index:
+        accumulate["index"] = item.index
+    if item.delta.content:
+        accumulate.setdefault("message", {})
+        accumulate["message"].setdefault("content", "")
+        accumulate["message"]["content"] += item.delta.content
+    if item.delta.tool_calls:
+        accumulate.setdefault("message", {})
+        accumulate["message"].setdefault("tool_calls", [])
+        for tool_call in item.delta.tool_calls:
+            if tool_call.id:
+                accumulate["message"]["tool_calls"].append({"id": tool_call.id, "type": "", "function": {"name": "", "arguments": ""}})
+            if tool_call.type:
+                accumulate["message"]["tool_calls"][-1]["type"] = tool_call.type
+            if tool_call.function and tool_call.function.name:
+                accumulate["message"]["tool_calls"][-1]["function"]["name"] = tool_call.function.name
+            if tool_call.function and tool_call.function.arguments:
+                accumulate["message"]["tool_calls"][-1]["function"]["arguments"] += tool_call.function.arguments
```

Comment:
We support parallel tool calls. So a model response can include an array of tool calls. Please log all of them, not only the  last one as per `[-1]'.